flight/targets/brain/fw/pios_board.c

```diff
@@ -531,79 +530,73 @@ void PIOS_Board_Init(void) {
-			(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_92) ? PIOS_MPU9250_ACCEL_LOWPASS_92_HZ : \
-			(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_41) ? PIOS_MPU9250_ACCEL_LOWPASS_41_HZ : \
-			(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_20) ? PIOS_MPU9250_ACCEL_LOWPASS_20_HZ : \
-			(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_10) ? PIOS_MPU9250_ACCEL_LOWPASS_10_HZ : \
-			(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_5) ? PIOS_MPU9250_ACCEL_LOWPASS_5_HZ : \
-			PIOS_MPU9250_ACCEL_LOWPASS_184_HZ;
-
-		int retval;
-		retval = PIOS_MPU9250_Init(pios_i2c_internal_id, PIOS_MPU9250_I2C_ADD_A0_LOW, use_internal_mag,
-									mpu9250_gyro_lpf, mpu9250_accel_lpf, &pios_mpu9250_cfg);
-		if (retval == -10)
-			PIOS_HAL_CriticalError(PIOS_LED_ALARM, PIOS_HAL_PANIC_IMU);
-		if (retval != 0)
-			PIOS_HAL_CriticalError(PIOS_LED_ALARM, PIOS_HAL_PANIC_IMU);
-
-		// To be safe map from UAVO enum to driver enum
-		uint8_t hw_gyro_range;
-		HwBrainGyroFullScaleGet(&hw_gyro_range);
-		switch(hw_gyro_range) {
-			case HWBRAIN_GYROFULLSCALE_250:
-				PIOS_MPU9250_SetGyroRange(PIOS_MPU60X0_SCALE_250_DEG);
-				break;
-			case HWBRAIN_GYROFULLSCALE_500:
-				PIOS_MPU9250_SetGyroRange(PIOS_MPU60X0_SCALE_500_DEG);
-				break;
-			case HWBRAIN_GYROFULLSCALE_1000:
-				PIOS_MPU9250_SetGyroRange(PIOS_MPU60X0_SCALE_1000_DEG);
-				break;
-			case HWBRAIN_GYROFULLSCALE_2000:
-				PIOS_MPU9250_SetGyroRange(PIOS_MPU60X0_SCALE_2000_DEG);
-				break;
-		}
+#if defined(PIOS_INCLUDE_MPU)
+	uint8_t hw_mpu9250_dlpf;
+	HwBrainMPU9250GyroLPFGet(&hw_mpu9250_dlpf);
+	uint16_t gyro_lpf = \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250GYROLPF_184) ? 184 : \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250GYROLPF_92) ? 92 : \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250GYROLPF_41) ? 41 : \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250GYROLPF_20) ? 20 : \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250GYROLPF_10) ? 10 : \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250GYROLPF_5) ? 5 : \
+		184;
+
+	HwBrainMPU9250AccelLPFGet(&hw_mpu9250_dlpf);
+	uint16_t accel_lpf = \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_460) ? 460 : \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_184) ? 184 : \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_92) ? 92 : \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_41) ? 41 : \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_20) ? 20 : \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_10) ? 10 : \
+		(hw_mpu9250_dlpf == HWBRAIN_MPU9250ACCELLPF_5) ? 5 : \
+		184;
+
+	pios_mpu_dev_t mpu_dev = NULL;
+	int retval;
+	retval = PIOS_MPU_I2C_Init(&mpu_dev, pios_i2c_internal_id, &pios_mpu_cfg);
+	if (retval == -10)
+		PIOS_HAL_CriticalError(PIOS_LED_ALARM, PIOS_HAL_PANIC_IMU);
+	if (retval != 0)
+		PIOS_HAL_CriticalError(PIOS_LED_ALARM, PIOS_HAL_PANIC_IMU);
+
+	PIOS_MPU_SetGyroBandwidth(gyro_lpf);
+	PIOS_MPU_SetAccelBandwidth(accel_lpf);
+
+	// To be safe map from UAVO enum to driver enum
+	uint8_t hw_gyro_range;
+	HwBrainGyroFullScaleGet(&hw_gyro_range);
+	switch(hw_gyro_range) {
+		case HWBRAIN_GYROFULLSCALE_250:
```

Comment:
For future reference, are case statements supposed to be indented or on the same column as the switch one?